### PR TITLE
vim-patch:8.1.{149,151,790,2100}

### PR DESCRIPTION
**Pull request by janlazo**
Close ~~https://github.com/neovim/neovim/pull/12345~~ and https://github.com/neovim/neovim/pull/13155

### CODE REVIEW

**Comment by janlazo** [comment]
cc @erw7 and @skippi

**Review by mhinz — APPROVED** [review]
I tested mksession on macOS with various complex sessions and everything sourced fine. Also explicitly tested for the empty buffer issue and used different combinations of `:[tl]cd`.

:+1:

**Review by erw7 — APPROVED** [review]

**Comment by janlazo** [comment]
```
[  ERROR   ] test/functional\ex_cmds\mksession_spec.lua @ 29: :mksession restores same :terminal buf in splits
test\helpers.lua:73: Expected objects to be the same.
Passed in:
(table: 0x14848030) {
 *[1] = 2
  [2] = 2
  [3] = 1 }
Expected:
(table: 0x14847fc0) {
 *[1] = 3
  [2] = 3
  [3] = 2 }
------------------------------------------------------------------------------
```
The actual result is correct when manually running the same commands. The vim patch is working as expected. This must have been a long-standing bug with mksession and terminal buffers.

File: test/functional/ex_cmds/mksession_spec.lua [hunk 1]
```diff
@@ -91,7 +91,7 @@ describe(':mksession', function()
     command('tabnext 1')
     eq(cwd_dir .. get_pathsep() .. tmpfile_base .. '1', funcs.expand('%:p'))
     command('tabnext 2')
-    eq(cwd_dir .. get_pathsep() .. tmpfile_base .. '2', funcs.expand('%:p'))
+    eq(cwd_dir .. '/' .. tmpfile_base .. '2', funcs.expand('%:p'))
```
Comment (janlazo) [inline test/functional/ex_cmds/mksession_spec.lua:94]: This is unexpected. `shellslash` is not set so `expand("%:p")` shouldn't use "/".

**Comment by janlazo** [comment]
```
2020-12-20T07:17:40.8116257Z Executed:  1804 Tests
2020-12-20T07:17:40.8116594Z  Skipped:   123 Tests
2020-12-20T07:17:40.8116898Z   FAILED:     1 Tests
2020-12-20T07:17:40.8117101Z 
2020-12-20T07:17:40.8117274Z 
2020-12-20T07:17:40.8117523Z Failures: 
2020-12-20T07:17:40.8117859Z 	From test_autocmd.vim:
2020-12-20T07:17:40.8118324Z 	Found errors in Test_autocmd_bufwipe_in_SessLoadPost():
2020-12-20T07:17:40.8119765Z 	function RunTheTest[37]..Test_autocmd_bufwipe_in_SessLoadPost line 24: Pattern 'E814' does not match '\nError detected while processing /home/runner/work/neovim/neovim/src/nvim/testdir/Session.vim:\nline  151:\nE325: ATTENTION'
2020-12-20T07:17:40.8120532Z 
```

Failed in TSAN and openbsd.

**Comment by janlazo** [comment]
That test consistently fails on my machine if including patch 8.1.0829. Excluding that patch should make all builds pass (except for the flaky tests on Windows and BSD environments).

**Pull request merged**